DataFormats/EgammaCandidates/interface/Photon.h

```diff
@@ -471,8 +471,53 @@ namespace reco {
     float pfMVA() const {return pfID_.mva;}
     // setters
     void setPflowIDVariables ( const PflowIDVariables& pfid ) {  pfID_ = pfid;}     
+    
+    struct PreshowerShapes { 
+      float effSigmaRR;
+    PreshowerShapes() : effSigmaRR(0.f) {}
+    };
+
+    void setPreshowerShapes( PreshowerShapes& shapes ) { preshower_shapes_ = shapes; }
+    const PreshowerShapes& preshowerShapes() const { return preshower_shapes_; }
+
+    struct ExtraShowerShapes {
+      float sigmaIetaIphi;
```

Comment:
as for electrons: why not add these to the ShowerShape struct?
I'd buy an "extra" if it was floating as a separate product that could be dropped  (as TrackExtra).